src/components/GlobalFilter/GlobalFilter.tsx

```diff
@@ -1,123 +1,176 @@
+  setWorkloads: ReturnType<typeof useSetAtom<typeof workloadsAtom>>
+) => {
   const navigate = useNavigate();
-  const registeredWith = useSelector(({ globalFilter: { scope } }: ReduxState) => scope);
-  const activeModule = useAtomValue(activeModuleAtom);
-  const isDisabled = useSelector(({ globalFilter: { globalFilterHidden } }: ReduxState) => globalFilterHidden || !activeModule);
-  const dispatch = useDispatch();
+  const registeredWith = useAtomValue(registeredWithAtom);
+  const isDisabled = useAtomValue(isDisabledAtom);
   return useCallback(
-    debounce((activeTags: any, search: any) => {
-      storeFilter(activeTags, hasAccess && !isDisabled, navigate);
-      batch(() => {
-        dispatch(
-          fetchAllTags({
+    debounce(async (activeTags: any, search: any) => {
+      // Set loading state to false before fetching
+      setTags((prev) => ({ ...prev, isLoaded: false }));
+      setSids((prev) => ({ ...prev, isLoaded: false }));
+      setWorkloads((prev) => ({ ...prev, isLoaded: false }));
+      try {
+        storeFilter(activeTags, hasAccess && !isDisabled, navigate);
+        await Promise.all([
+          getAllTags({
             registeredWith,
             activeTags,
             search,
-          })
-        );
-        dispatch(
-          fetchAllSIDs({
+          }),
+          getAllSIDs({
             registeredWith,
             activeTags,
             search,
-          })
-        );
-        dispatch(
-          fetchAllWorkloads({
+          }),
+          getAllWorkloads({
             registeredWith,
             activeTags,
             search,
-          })
-        );
-      });
+          }),
+        ]);
+      } catch (error) {
+        console.error('Failed to load global filter tags:', error);
+      } finally {
+        // Set loaded property to true even if an error occurs
+        setTags((prev) => ({ ...prev, isLoaded: true }));
+        setSids((prev) => ({ ...prev, isLoaded: true }));
+        setWorkloads((prev) => ({ ...prev, isLoaded: true }));
+      }
     }, 600),
     [registeredWith, hasAccess]
   );
 };
 
 const GlobalFilter = ({ hasAccess }: { hasAccess: boolean }) => {
   const [isOpen, setIsOpen] = useState(false);
-  const dispatch = useDispatch();
-  const isLoaded = useSelector(({ globalFilter: { tags, sid, workloads } }: ReduxState) => tags.isLoaded && sid.isLoaded && workloads.isLoaded);
-  const { count, total, tags, sid, workloads } = useSelector(
-    ({ globalFilter: { tags, sid, workloads } }: ReduxState) => ({
-      count: (tags.count || 0) + (sid.count || 0) + (workloads.count || 0),
-      total: (tags.total || 0) + (sid.total || 0) + (workloads.total || 0),
-      tags: tags.items || [],
-      sid: sid.items || [],
-      workloads,
-    }),
-    shallowEqual
-  );
+  const isLoaded = useAtomValue(isLoadedAtom);
+  const tagsData = useAtomValue(tagsAtom);
+  const sidsData = useAtomValue(sidsAtom);
+  const workloadsData = useAtomValue(workloadsAtom);
```

Comment:
Here I would create a common "read atom" that would give all of these. WE don't have to retrieve them individually, and the re-render count will decrease.